tempodb/backend/block_meta.go

```diff
@@ -4,52 +4,158 @@ import (
 	"bytes"
 	"time"
 
+	"github.com/cespare/xxhash/v2"
 	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/traceql"
+)
+
+// DedicatedColumnType is the type of the values in the dedicated attribute column. Only 'string' is supported.
+type DedicatedColumnType string
```

Comment:
Both enumerations have different representations (string vs. int) and a different set of possible values

Reply:
> Both enumerations have different representations (string vs. int) 

Int would technically be more efficient as we're passing it everywhere. The big cost would be human readability (which I value). Maybe we just need to get this change in some scale testing and see how it does.

> and a different set of possible values

I would actually prefer the using the traceql enums for this reason. Since they represent all the types and scopes possible to put into dedicated columns. Then we can just fail validation if an unsupported scope/type is chosen.

Reply:
TraceQL types are different than the underlying otel attribute types.  I think it makes sense to keep them separate. i.e. TraceQL isn't tempo's native type system, the otel proto is.  There was some discussion on the original PR here: https://github.com/grafana/tempo/pull/2592#discussion_r1243794127

Reply:
Thanks for the history. Works for me :+1: